internal/compiler/passes/resolving.rs

```diff
@@ -77,7 +77,11 @@ fn resolve_expression(
                 Expression::Invalid
             }
         };
-        *expr = new_expr;
+        match expr {
+            // TODO: Handle nested debug hooks? Probably not going to happen anyway...
```

Comment:
Maybe add an assert?

Reply:
Actually I need to remove that: They can not be nested now that I removed the parser that allowed users to create debug hooks.